### PR TITLE
[REFACTOR][RELAY] move fallback_device to config

### PR DESCRIPTION
**Pull request by zhiics**
Per #5650, this PR moves this fallback_device field in `PassContext` into config. 

@tqchen @junrushao1994

### CODE REVIEW

File: tests/python/relay/test_pass_annotation.py [hunk 1]
```diff
@@ -538,9 +539,10 @@ def expected():
     expected_index = [2, 2, 2, 1, 1, 1, 2, 2]
     check_annotated_graph(annotated_func, expected_func)
     params = {"a": a_data, "b": b_data, "c": c_data, "d": d_data}
-    config = {"opt_level": 0}
-    config["fallback_device"] = fallback_device
-    with relay.build_config(**config):
+    fall_dev = tvm.tir.IntImm("int32", fallback_device.device_type)
+    with tvm.transform.PassContext(opt_level=0,
+                                   config={"relay.fallback_device_type":
+                                           tvm.tir.IntImm(32, fall_dev)}):
```
Comment (tqchen) [inline tests/python/relay/test_pass_annotation.py:545]: do we ned the intimm? I feel that conversion from int to the intImm is automatic

Comment (zhiics) [reply tests/python/relay/test_pass_annotation.py:545]: yeah, it's not necessary. Let me update.

File: tests/cpp/relay_transform_sequential.cc [hunk 2]
```diff
@@ -70,7 +70,7 @@ TEST(Relay, Sequential) {
   auto mod = IRModule::FromExpr(func);
   auto pass_ctx = relay::transform::PassContext::Create();
   pass_ctx->opt_level = 3;
-  pass_ctx->fallback_device = 1;
+  pass_ctx->config.Set("relay.fallback_device_type", IntImm(DataType::Int(32), 1));
```
Comment (tqchen) [inline tests/cpp/relay_transform_sequential.cc:73]: After merging the code, I find that we can further simplify it to pass_ctx->config.Set("relay.fallback_device_type", Integer(1));

Comment (zhiics) [reply tests/cpp/relay_transform_sequential.cc:73]: Yeah, I tried this but the compilation failed. I can take a deeper look though.

**Pull request merged**